build.nix

```diff
@@ -36,11 +36,15 @@
   #  Which drops the run-time dependency on the crates-io source thereby
   #  significantly reducing the Nix closure size.
 , removeReferencesToSrcFromDocs
+, checkOnly ? false
```

Comment:
It looks like it makes sense to only have one of those active at once, so IMO we should either have an extra assertion:

```
if (checkOnly && (testOnly || clippyOnly)) ||
    (testOnly && (checkOnly || clippyOnly)) ||
    (clippyOnly && (checkOnly || testOnly))
  builtins.throw "Only one of `checkOnly`, `testOnly`, `clippyOnly` can be active."
```

... or switch to an enum-like approach:

```
# ...
, mode ? "build" # or `check`, or `test`, or `clippy`
# ...

if !builtins.elem mode ["build", "check", "test", "clippy"]
  builtins.throw "Unknown mode ${mode}, allowed modes: build, check, test, clippy"
```

Reply:
Switch to a `mode` "enum".